api/search.go

```diff
@@ -109,65 +108,46 @@ func paramGetBool(params url.Values, key string, defaultValue bool) bool {
 
 // CreateRequests reads the parameters from the request and generates the corresponding SearchRequest and CountRequest
 // If any validation fails, the http.Error is already handled, and nil is returned: in this case the caller may return straight away
-func CreateRequests(w http.ResponseWriter, req *http.Request, validator QueryParamValidator, nlpCriteria *query.NlpCriteria) (string, *query.SearchRequest, *query.CountRequest) {
+func CreateRequests(w http.ResponseWriter, req *http.Request, cfg *config.Config, validator QueryParamValidator, nlpCriteria *query.NlpCriteria) (string, *query.SearchRequest, *query.CountRequest) {
 	ctx := req.Context()
 	params := req.URL.Query()
 
-	q := params.Get(ParamQ)
-	sanitisedQuery := sanitiseDoubleQuotes(q)
-	queryHasSpecialChars := checkForSpecialCharacters(sanitisedQuery)
-
-	if queryHasSpecialChars {
-		log.Info(ctx, "rejecting query as contained special characters", log.Data{"query": sanitisedQuery})
-		http.Error(w, "Invalid characters in query", http.StatusBadRequest)
+	// Sanitize and validate the query string
+	sanitisedQuery, err := sanitizeAndValidateQuery(ctx, w, params)
```

Comment:
```suggestion
	sanitisedQuery, err := sanitiseAndValidateQuery(ctx, w, params)
```
inconsistent spelling (`z` in only 4 places)